### PR TITLE
Add action to list repositories

### PR DESCRIPTION
**Pull request by natete**

### CODE REVIEW

File: .github/actions/list-repos/action.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
+name: List Repositories
+description: Get the list of repositories
+
+inputs:
+  private_key:
+    required: true
+    description: Password to access GitHub
```
Comment (sgmoratilla) [inline .github/actions/list-repos/action.yml:7]: It's not a password haha It's a private key.

File: .github/actions/list-repos/action.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  regex_filter:
+    required: false
+    description: A regular expresion to filter repositories
```
Comment (sgmoratilla) [inline .github/actions/list-repos/action.yml:10]: to filter in

File: .github/actions/list-repos/action.yml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    required: false
+    description: A list of repos to exclude from the response
+  exclude_archived:
+    required: false
+    description: Whether to exclude archived repositories from the response or not
+    default: 'false'
+
+outputs:
+  repos:
+    description: The list of repos
```
Comment (sgmoratilla) [inline .github/actions/list-repos/action.yml:21]: I would link here to the format of the repo object in Github doc (if there is any). If not, i would clarify that it is not the name of the repo, but the repo object.

Comment (natete) [reply .github/actions/list-repos/action.yml:21]: But I'm returning just an array of repo names here. The last step of the script does the trick

File: .github/actions/list-repos/action.yml [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+          
+          if (excludedRepos) {
+            repositories = repositories.filter(repo => !excludedRepos.includes(repo.name));          
+          }
+          
+          if (${{inputs.exclude_archived}}) {
+            repositories = repositories.filter(repo => !repo.archived);
+          }
+
+          return repositories.map(repo => repo.name);
```
Comment (sgmoratilla) [inline .github/actions/list-repos/action.yml:59]: btw, should we return just the names of the whole object?

If this is a generic action, maybe it's more flexible if we return the object.

**Review by sgmoratilla — APPROVED** [review]

**Pull request merged**